fuel-p2p/src/behavior.rs

```diff
@@ -0,0 +1,95 @@
+use std::{
+    collections::VecDeque,
+    task::{Context, Poll},
+};
+
+use libp2p::{
+    gossipsub::{Gossipsub, GossipsubEvent, TopicHash},
+    identify::IdentifyEvent,
+    identity::Keypair,
+    swarm::{
+        NetworkBehaviour, NetworkBehaviourAction, NetworkBehaviourEventProcess, PollParameters,
+    },
+    NetworkBehaviour, PeerId,
+};
+
+use crate::discovery::{DiscoveryBehaviour, DiscoveryEvent};
+
+#[derive(Debug)]
+pub enum FuelBehaviourEvent {
+    GossipsubMessage {
+        peer_id: PeerId,
+        topic: TopicHash,
+        message: Vec<u8>,
+    },
+}
+
+#[derive(NetworkBehaviour)]
+#[behaviour(
+    out_event = "FuelBehaviourEvent",
+    poll_method = "poll",
+    event_process = true
+)]
+pub struct FuelBehaviour {
+    /// message propagation for p2p
+    gossipsub: Gossipsub,
+
+    /// node discovery
+    discovery: DiscoveryBehaviour,
+
+    //peer info let's do it later
+    //identify: Identify,
+    #[behaviour(ignore)]
+    events: VecDeque<FuelBehaviourEvent>,
```

Comment:
q: should this be unbounded or a reasonable limit should be enforced? (i.e. how many messages each peer can send us)